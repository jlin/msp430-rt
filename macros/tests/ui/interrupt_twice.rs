#![no_main]
#![feature(abi_msp430_interrupt)]

use msp430_rt_macros::{entry, interrupt};

#[entry]
fn main() -> ! {
    loop {}
}

#[allow(non_camel_case_types)]
enum interrupt {
    TIM1,
    TIM2,
}

#[interrupt]
fn TIM1() {
    // Different name, so it shouldn't be flagged
    loop {}
}

#[interrupt]
fn TIM2() {
    loop {}
}

#[interrupt]
fn TIM2() {
    loop {}
}
