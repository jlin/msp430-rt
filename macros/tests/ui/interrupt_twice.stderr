error[E0428]: the name `TIM2` is defined multiple times
  --> tests/ui/interrupt_twice.rs:28:1
   |
23 | #[interrupt]
   | ------------ previous definition of the value `TIM2` here
...
28 | #[interrupt]
   | ^^^^^^^^^^^^ `TIM2` redefined here
   |
   = note: `TIM2` must be defined only once in the value namespace of this module
   = note: this error originates in the attribute macro `interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0570]: `"msp430-interrupt"` is not a supported ABI for the current target
  --> tests/ui/interrupt_twice.rs:17:1
   |
17 | #[interrupt]
   | ^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0570]: `"msp430-interrupt"` is not a supported ABI for the current target
  --> tests/ui/interrupt_twice.rs:23:1
   |
23 | #[interrupt]
   | ^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0570]: `"msp430-interrupt"` is not a supported ABI for the current target
  --> tests/ui/interrupt_twice.rs:28:1
   |
28 | #[interrupt]
   | ^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)
